### ISSUE TITLE
Use Angle by default, not only on Mac

### ISSUE THREAD

**Issue by Nathan-MV** [mkxp-z/mkxp-z#66]
@Eblo implementation can be useful

**Comment by Eblo** [comment]
Yeah, this is a thing I did for my fork. I thought of submitting a PR to this repo, but my changes have some specifics that one may want massaged for general use. This involved fiddling with SDL to get it to use a locally-provided lib on Linux, as opposed to trying the `/usr/bin` variant first. I also just lazily stole the lib files from Ungoogled Chromium instead of building them myself. The commits for each lib do not match between MacOS, Linux, and Windows. Ideally all of them would be built from the same commit for parity. Another problem is that I only include the prebuilt libs for Vulkan and GLES (to match what the MacOS build does), but there are other options like SwiftShader to consider.

The fun of this is that it allows for easily switching between rendering backends like Vulkan and OpenGL. Further notes:
* ANGLE Vulkan on Linux Mint seems to fix various issues that otherwise use settings like `subImageFix` and `enableBlitting`. i.e. Vulkan removes the need for those settings.
* ~~ANGLE Vulkan on Windows actually has a new issue with the right (or left?) edge of an individual tile flickering as they scroll horizontally. D3D11 does not produce this issue, nor does Vulkan on Linux.~~ This seems to be some sort of precision issue that can be fixed by modifying the tilemapvx vertex shader. See https://github.com/Eblo/mkxp-z-eulogy/commit/8ed347b49968998691ad251e9cb2e6344b22143e
* The aforementioned SDL hack for Linux. I forked mkxp-z's SDL to add a branch for Linux to change the checked location for `libEGL.so`. Without this, mkxp-z will check `/usr/bin` for the non-ANGLE variant and use that.
* Vulkan does not work on a VM.
* Using ANGLE for all three OSes gives parity.

Here are the links. The ANGLE readme mentions the commits used for the included lib files for each OS:
[Windows ANGLE commit](https://github.com/Eblo/mkxp-z-eulogy/commit/a8ded9de7d02e63ddb3e5cdee4b437ef480defd5)
[Linux ANGLE commit](https://github.com/Eblo/mkxp-z-eulogy/commit/8e141785d52fd677212b7119c25aa27c7fd45ad6)
[SDL hack for ANGLE on Linux](https://github.com/Eblo/SDL/commit/170dbe4a9d8a4887b1ffc9428f4109f4d953cf2a)
